cmd/influxd/run/server.go

```diff
@@ -290,6 +290,7 @@ func (s *Server) appendRetentionPolicyService(c retention.Config) {
 	srv := retention.NewService(c)
 	srv.MetaClient = s.MetaClient
 	srv.TSDBStore = s.TSDBStore
+	srv.DropShardMetaRef = retention.OSSDropShardMetaRef(s.MetaClient)
```

Comment:
Can this assignment just be `s.MetaClient.DropShard` and we eliminate the wrapper function?

Reply:
Enterprise has to use a different method for `DropShardMetaRef` (`MetaClient.RemoveShardOwner` to be precise)